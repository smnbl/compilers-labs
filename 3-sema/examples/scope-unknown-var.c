// RUN-WITH-ARGS: -dump-symbol-table

int foo(int arg0, int arg1)
{
    int var0;
    int var1;
    int var2;

    return does_not_exist;
}
