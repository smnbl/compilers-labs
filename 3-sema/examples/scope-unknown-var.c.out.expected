sema: error: Undefined variable 'does_not_exist'
└── Program <9>
    └── FuncDecl: returnType = 'int', name = 'foo' <8>
        ├── VarDecl: type = 'int', name = 'arg0' <0>
        ├── VarDecl: type = 'int', name = 'arg1' <1>
        └── CompoundStmt <7>
            ├── VarDecl: type = 'int', name = 'var0' <2>
            ├── VarDecl: type = 'int', name = 'var1' <3>
            ├── VarDecl: type = 'int', name = 'var2' <4>
            └── ReturnStmt <6>
                └── VarRefExpr: name = 'does_not_exist' <5>
